woocommerce/woocommerce-ios
# Bump deployment target to 16.0

Closes: #11867 


## Description:
This PR updates the minimum deployment target to be iOS 16. There's ~10 deprecation warnings regarding NavigationLinks that haven't been addressed since these will need additional work. Details below.

## Changes:
* Updated all targets to iOS 16
* Updated deprecated `Locale.regionCode?` to `Locale.region?.identifier`
* Addressed some lint warnings that -oddly- didn't show previously, regarding whitespace, or 160+chars usage per line.

## Not addressed:
A CUW P2 can be created for these:

**1. NavigationLink deprecation warnings:** 

We use these in the `LazyNavigationLink` component, and in several views:`InPersonPaymentsMenu`, , `PaymentMethodsView`, `PaymentsRow`, and `ShippingLabelPackageList`. 

The update generally requires to wrap the existing views in either a `NavigationStack` or a `NavigationSplitView`, which modifies the current navigation behaviour. In some cases (Payments Menu) it might be necessary to implement some sort of navigation routing for each navigation case. In order to be sure that the different rows and destinations work correctly, as well as the different IPP flows that rely on account state changes work as expected, we should do this separately from this PR and assure we're not introducing any bug.

**2. Some CoreTelephony properties been deprecated with no replacement:**
 
We use these in the "network information" fields used for Zendesk when a user sends a ticket, specifically:
`serviceSubscriberCellularProviders`, `carrierName`, `isoCountryCode`. From my research, these have been deprecated for privacy reasons.

## Testing instructions
- CI tests should pass
- Go to Menu > Site switcher > `+ Add a Store` > `Create New Store` > Follow the creation flow. Observe how a predefined country is recommended during creation.

---
